tests/test-full-stack.py

```diff
@@ -5,7 +5,7 @@
 def generate_aiidalab_install_output(aiidalab_exec, nb_user):
     def _generate_aiidalab_install_output(package_name):
         output = (
-            aiidalab_exec(f"aiidalab install --yes {package_name}", user=nb_user)
+            aiidalab_exec(f"aiidalab install --yes --pre {package_name}", user=nb_user)
```

Comment:
Not sure if this is needed, since we're also installing directly from the branch in `test_install_apps_from_default_branch`. But I guess it is fine.

Reply:
Much appreciate for writing the detailed test flow, I'll test on QeApp with @superstar54 who has quite many containers with lots of calculations. 

Per `--pre` options I added, it is for QeApp specifically since we didn't make the release `v23.10.0` yet but the prerelease version. The stable version is still `v23.04.6` which pinning to `aiida-core~=2.3.0` will try to downgrade aiida-core and fail the test (exactly what we improved in this PR). 

I personally think here it is better to not use `--pre` but only test the stable version. I will keep this PR until we make the release of `v23.10.0` which the planned date is 6th November.

Reply:
@unkcpz @superstar54 friendly ping, would be good to test and merge so that others can test this via the `edge` image.

Reply:
Tested, all good. The migration is run and calculations can be reloaded with showing the same result properly.